test/one/profiler/test/Runner.java

```diff
@@ -186,6 +186,12 @@ public static void main(String[] args) throws Exception {
         for (RunnableTest rt : allTests) {
             long start = System.nanoTime();
             TestResult result = run(rt, decl);
+            
+            if (result.status() == TestStatus.FAIL && currentOs == Os.MACOS) {
```

Comment:
This does not make distinction between macos-x64 and macos-arm that we might want.
My idea was to have the number of retries externally configurable (default == 0). We will pass it as `1` only for macos-x64 job in GHA.